### PR TITLE
Refactor models

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

**Comment by stefanv** [comment]
The real test will be against SkyPortal, I suppose.

File: app/handlers/base.py [hunk 1]
```diff
@@ -1,11 +1,16 @@
+__all__ = ['BaseHandler']
```
Comment (acrellin) [inline app/handlers/base.py:1]: This shouldn't be necessary as we never do `from baselayer.app.handlers.base import *`

File: app/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from .access import *
+from .acl import ACL, Role
+from .base import *
+from .cron_job_run import CronJobRun
+from .join import JoinModel, join_model
+from .session import *
+from .token import *
+from .user import User
+from .util import *
```
Comment (acrellin) [inline app/models/__init__.py:9]: Why not use a consistent pattern here -- either import each item explicitly, or always use `*`?

File: app/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Define these here to prevent circular imports elsewhere
+RoleACL = join_model("role_acls", Role, ACL)
+RoleACL.__doc__ = "Join table class mapping Roles to ACLs."
+
+UserACL = join_model("user_acls", User, ACL)
+UserACL.__doc__ = "Join table mapping Users to ACLs"
+UserRole = join_model("user_roles", User, Role)
+UserRole.__doc__ = "Join table mapping Users to Roles."
+
+del access, acl, base, cron_job_run, join, session, token, user, util
```
Comment (acrellin) [inline app/models/__init__.py:20]: Why is this line needed?

Comment (stefanv) [reply app/models/__init__.py:20]: Probably because of the `*` imports that I should remove above.

**Comment by guynir42** [comment]
@stefanv is this done? I think on SP it was completed a while ago...

**Comment by stefanv** [comment]
No, we ran into difficulties we could not debug at the time. Will have to re-attempt.
